pkg/controller/node/node_controller_test.go

```diff
@@ -429,129 +429,63 @@ func TestGetCandidateMachines(t *testing.T) {
 		},
 		expected: nil,
 	}, {
-		//progress all available nodes
+		// node-2 is going to change config, so we can only progress one more
 		progress: 2,
 		nodes: []*corev1.Node{
 			newNodeWithReady("node-0", "v1", "v1", corev1.ConditionTrue),
-			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionFalse),
+			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionTrue),
 			newNodeWithReady("node-2", "v0", "v1", corev1.ConditionTrue),
 			newNodeWithReady("node-3", "v0", "v0", corev1.ConditionTrue),
 			newNodeWithReady("node-4", "v0", "v0", corev1.ConditionTrue),
 		},
-		expected: []*corev1.Node{newNodeWithReady("node-3", "v0", "v0", corev1.ConditionTrue), newNodeWithReady("node-4", "v0", "v0", corev1.ConditionTrue)},
+		expected: []string{"node-3"},
 	}, {
-		//progress nodes but limited
+		// We have a node working, don't start anything else
 		progress: 1,
 		nodes: []*corev1.Node{
 			newNodeWithReady("node-0", "v1", "v1", corev1.ConditionTrue),
-			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionFalse),
+			newNodeWithReady("node-1", "v1", "v1", corev1.ConditionTrue),
 			newNodeWithReady("node-2", "v0", "v1", corev1.ConditionTrue),
 			newNodeWithReady("node-3", "v0", "v0", corev1.ConditionTrue),
 			newNodeWithReady("node-4", "v0", "v0", corev1.ConditionTrue),
 		},
-		expected: []*corev1.Node{newNodeWithReady("node-3", "v0", "v0", corev1.ConditionTrue)},
+		expected: nil,
 	}, {
 		//progress on old stuck node
 		progress: 1,
 		nodes: []*corev1.Node{
 			newNodeWithReady("node-0", "v1", "v1", corev1.ConditionTrue),
-			newNodeWithReady("node-1", "v0.1", "v0.2", corev1.ConditionFalse),
+			newNodeWithReadyAndDaemonState("node-1", "v0.1", "v0.2", corev1.ConditionTrue, daemonconsts.MachineConfigDaemonStateDegraded),
```

Comment:
This case also changed to differentiate on "non-working" MCD state - we do want to reset annotations on nodes that have failed on an update and aren't targeting our current config.